prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -283,6 +283,30 @@ def map_via_neighbor_voltages(lines, neighbors, func, method_name):
     lines["VOLTAGE"] = lines["VOLTAGE"].astype(float)
 
 
+def create_buses(lines):
+    """Using line and substation information, create a bus within each substation and a
+    mapping of buses to substations.
+
+    :param pandas.DataFrame lines: data frame of lines.
+    :return: (*tuple*) --
+        (*pandas.Series*) -- mapping of buses to voltages.
+        (*pandas.Series*) -- mapping of buses to substations.
+    """
+    sub_1_lines = pd.Series(lines.groupby("SUB_1_ID").groups).apply(set)
+    sub_2_lines = pd.Series(lines.groupby("SUB_2_ID").groups).apply(set)
+    sub_lines = sub_1_lines.combine(sub_2_lines, set.union, fill_value=set())
+    # Get the set of voltages for any line(s) connected to each substation
+    substation_voltages = sub_lines.map(lambda x: sorted(set(lines.VOLTAGE.loc[x])))
+    buses = substation_voltages.explode()
+    buses = buses.loc[~buses.isna()]
```

Comment:
Why we need this? `buses` won't have `NaN` after an `explode` operation, no?

Reply:
It can if an empty list is exploded.

Reply:
If every line has voltage augmented, there won't be any empty list, right?

Reply:
If we add the new functionality from #216 into `build_transmission` before we call the voltage augmentation, and before this function, you're right, but we're not doing that yet.